coffea/processor/executor.py

```diff
@@ -1823,12 +1832,21 @@ def run(
 
         executor = self.executor.copy(**exe_args)
 
-        wrapped_out, e = executor(chunks, closure, None)
-        if wrapped_out is None:
+        def check_chunks(x):
+            if meta:
+                return next(x, None) is not None
```

Comment:
I think this might be a problem. The first element of the generator will be lost?

Reply:
I wonder if it is better to simply check the len() == 0 of   `chunks_to_count` above (line 1811). It leaves `chunks` alone if it comes from a generator, and it has the same contents as chunks if not. Further, you may not need to define the function `check_chunks` anymore.

Reply:
Yes thats a much better idea.  It also separates the lack of chunks from checking bad processors. Thanks! I implemented and `pytest` passes at least.

Reply:
If the first element is lost (which it should, you're right) the tests should fail ... makes me think the test is not covering something.

Reply:
https://github.com/MoAly98/coffea/blob/391378089f3f30337205a293801040b09e606457/coffea/processor/executor.py#L1802

This is what's breaking. You can still get `chunks_to_count`/`chunks` as a generator. I guess a `check_chunks` functio there could make sense...